R/create_trends_ensemble.R

```diff
@@ -0,0 +1,158 @@
+#' Generate predictions for the trends ensemble, a quantile median of component
+#' baseline models
+#'
+#' @param component_variations a `data.frame` where each row specifies a set of
+#'   hyperparameters to use for a single baseline model fit, with columns
+#'   `transformation`, `symmetrize`, `window_size`, and `temporal_horizon`.
+#'   See details for more information
+#' @param target_ts a `data.frame` of target data in a time series format
+#'   (contains columns `time_index`, `location`, and `observation`) for a single
+#'   location
+#' @param reference_date string of the reference date for the forecasts, i.e.
+#'   the date relative to which the targets are defined (usually Saturday for
+#'   weekly targets). Must be in the ymd format, with yyyy-mm-dd format recommended.
+#' @param horizons numeric vector of prediction horizons relative to
+#'   the reference_date, e.g. 0:3 or 1:4, and interpreted to be in terms of the
+#'   same temporal resolution as the provided `target_ts`.
+#' @param target character string specifying the name of the prediction target
+#' @param quantile_levels numeric vector of quantile levels to output; set to NULL
+#'   if quantile outputs not requested. Defaults to NULL.
+#' @param n_samples integer of amount of samples to output (and predict);
+#'   set to NULL if sample outputs not requested (in this case 100000 samples
+#'   are generated from which to extract quantiles). Defaults to NULL.
```

Comment:
I have a vague recollection that we were going to introduce two parameters: one for a number of samples to output if the sample output is requested, and a separate one for the number of samples generated behind the scenes for purposes of extracting quantiles.  Does that seem right/do you remember discussion about this?

Reply:
I see that there is an `n_sim` argument in `get_baseline_predictions` that allows us to disambiguate here.  It seems like it would make sense to put a similar argument here, right?